plugins/woocommerce/src/Internal/StockNotifications/Notification.php

```diff
@@ -0,0 +1,371 @@
+	 */
+	public function get_date_confirmed( $context = 'view' ) {
+		return $this->get_prop( 'date_confirmed', $context );
+	}
+
+	/**
+	 * Get the date last attempt.
+	 *
+	 * @param string $context Context.
+	 * @return \WC_DateTime|null Datetime object if the date is set or null if there is no date.
+	 */
+	public function get_date_last_attempt( $context = 'view' ) {
+		return $this->get_prop( 'date_last_attempt', $context );
+	}
+
+	/**
+	 * Get the date notified.
+	 *
+	 * @param string $context Context.
+	 * @return \WC_DateTime|null Datetime object if the date is set or null if there is no date.
+	 */
+	public function get_date_notified( $context = 'view' ) {
+		return $this->get_prop( 'date_notified', $context );
+	}
+
+	/**
+	 * Get the date cancelled.
+	 *
+	 * @param string $context Context.
+	 * @return \WC_DateTime|null Datetime object if the date is set or null if there is no date.
+	 */
+	public function get_date_cancelled( $context = 'view' ) {
+		return $this->get_prop( 'date_cancelled', $context );
+	}
+
+	/**
+	 * Get the cancellation source.
+	 *
+	 * @param string $context Context.
+	 * @return string|null The cancellation source or null if there is no source.
+	 */
+	public function get_cancellation_source( $context = 'view' ) {
+		return $this->get_prop( 'cancellation_source', $context );
+	}
+
+	/**
+	 * Get the product.
+	 *
+	 * @return \WC_Product|false
+	 */
+	public function get_product() {
+		if ( ! empty( $this->product ) ) {
+			return $this->product;
+		}
+
+		$product = wc_get_product( $this->get_prop( 'product_id' ) );
+		if ( ! $product ) {
+			return false;
+		}
+
+		$this->product = $product;
+		return $product;
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| Setters
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Set the product ID.
+	 *
+	 * @param int $product_id Product ID.
+	 */
+	public function set_product_id( $product_id ) {
+
+		// Reset runtime cache if the product ID has changed.
+		if ( is_a( $this->product, 'WC_Product' ) && $product_id !== $this->product->get_id() ) {
+			$this->product = null;
+		}
+		$this->set_prop( 'product_id', $product_id );
+	}
+
+	/**
+	 * Set the user ID.
+	 *
+	 * @param int $user_id User ID.
+	 */
+	public function set_user_id( $user_id ) {
+		$this->set_prop( 'user_id', $user_id );
+	}
+
+	/**
+	 * Set the user email.
+	 *
+	 * @param string $user_email User email.
+	 */
+	public function set_user_email( $user_email ) {
+		$this->set_prop( 'user_email', $user_email );
+	}
+
+	/**
+	 * Set the status.
+	 *
+	 * @param string $status Status.
+	 */
+	public function set_status( $status ) {
+
+		if ( ! in_array( $status, NotificationStatus::get_valid_statuses(), true ) ) {
+			// Default to pending.
+			$status = NotificationStatus::PENDING;
+		}
+
+		$this->set_prop( 'status', $status );
+	}
+
+	/**
+	 * Set the date created.
+	 *
+	 * @param string $date_created Date created.
```

Comment:
The type here says `string`, but it will accept other types and the datastore specifically uses `time() :int`. Can we update the docblocks to reflect that accepted types?

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/57062/commits/5667dde7e83f2c5b66f2ba5afc50935b76dfac7a